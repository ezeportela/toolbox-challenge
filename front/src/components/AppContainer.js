import React from "react";
import { useSelector } from "react-redux";
import Container from "react-bootstrap/Container";
import Spinner from "react-bootstrap/Spinner";

const AppContainer = ({ children }) => {
  const { loading } = useSelector((state) => state);

  return (
    <Container className="mt-7 mb-5">
      {loading && (
        <div className="mt-5 mb-5 d-flex justify-content-center">
          <Spinner animation="border" role="status">
            <span className="visually-hidden">Loading...</span>
          </Spinner>
        </div>
      )}

      {children}
    </Container>
  );
};

export default AppContainer;
