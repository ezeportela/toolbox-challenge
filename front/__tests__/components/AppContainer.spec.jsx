import React from "react";
import { Provider, useSelector } from "react-redux";
import configureStore from "redux-mock-store"; // If you use redux-mock-store to create a mock store
import renderer from "react-test-renderer";
import AppContainer from "../../src/components/AppContainer";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn().mockImplementation((state) => state),
}));

describe("AppContainer (Snapshot)", () => {
  it("Should render component", () => {
    const mockStore = configureStore([]);
    const initialState = {
      loading: true,
      files: [],
    };
    useSelector.mockReturnValue(initialState);

    const store = mockStore(initialState);

    const component = renderer
      .create(
        <Provider store={store}>
          <AppContainer />
        </Provider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
