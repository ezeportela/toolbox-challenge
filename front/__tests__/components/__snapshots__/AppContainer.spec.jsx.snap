// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer (Snapshot) Should render component 1`] = `
<div
  className="mt-7 mb-5 container"
>
  <div
    className="mt-5 mb-5 d-flex justify-content-center"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
